### PR TITLE
1.0.0/feature/auto rotation

### PR DESCRIPTION
**Pull request by zeppone**
add auto rotation (:

### CODE REVIEW

File: src/load_Image/auto_rotation.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        GdkPixbuf *newpixbuf = gdk_pixbuf_new_subpixbuf (pixbuf, 0, 0,gdk_pixbuf_get_width (pixbuf), gdk_pixbuf_get_height (pixbuf));
+        temp_image = gtk_image_new_from_pixbuf (newpixbuf);
+
+        int var = get_variance(temp_image, i);
+        if (var > var_max) {
+            var_max = var;
+            degree_max = i;
+        }
+        printf("degree: %d\n",i);
+        printf("variance: %d\n",var);
```
Comment (Winbay) [inline src/load_Image/auto_rotation.c]: no print

File: src/load_Image/auto_rotation.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    for (int i = 0; i < height; ++i) {
+        average += hp[i];
+    }
+    average = average / height;
+    int var;
+    for (int i = 0; i < height; ++i) {
+        var += (hp[i] - average) * (hp[i] - average);
+    }
+    var = var / height;
+    free(hp);
```
Comment (Winbay) [inline src/load_Image/auto_rotation.c:21]: why free ? there is no allocation ?

File: src/load_Image/auto_rotation.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#ifndef UNREAD_AUTO_ROTATION_H
+#define UNREAD_AUTO_ROTATION_H
```
Comment (Winbay) [inline src/load_Image/auto_rotation.h:2]: prefer `UNREAD__SRC__LOAD_IMAGE__AUTO_ROTATION_H`

File: src/load_Image/auto_rotation.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+void auto_rotation(GtkWidget *image) {
+    GtkWidget *temp_image;
+    GdkPixbuf *pixbuf;
+    int degree_max;
+    int var_max = 0;
+
+    for (int i = -30; i <= 30; i = i + 2) {
+        //getting a temporary image for test
+        pixbuf = gtk_image_get_pixbuf ((GtkImage*) image);
+        GdkPixbuf *newpixbuf = gdk_pixbuf_new_subpixbuf (pixbuf, 0, 0,gdk_pixbuf_get_width (pixbuf), gdk_pixbuf_get_height (pixbuf));
```
Comment (Winbay) [inline src/load_Image/auto_rotation.c]: more than 80 column

File: src/load_Image/segmentation.c [hunk 5]
```diff
@@ -57,7 +58,6 @@ int * verticalProjection(GdkPixbuf *pixbuf)
             p = which_pixels + y * rowstride + x * n_channels;
             // image coming from transform_graylvl.c are either totally black or totally white,
             if (p[0] < 128) {
-
                 vp[x] += 1;
             }
         }
```
Comment (Winbay) [inline src/load_Image/segmentation.c:63]: free vp ?

File: src/load_Image/segmentation.c [hunk 6]
```diff
@@ -131,6 +131,46 @@ void lineSegmentation(GdkPixbuf *pixbuf)
```
Comment (Winbay) [inline src/load_Image/segmentation.c:136]: free all malloc ?

**Review by JulesdeCube — DISMISSED** [review]

**Pull request merged**